types.go

```diff
@@ -0,0 +1,55 @@
+package process
```

Comment:
Will there be any combined interfaces, like `StoppableProcess` similar to `io.ReadWriter`? Or will a process just be `Process` to users? It might be useful to have `InitializableStoppableProcess`, `InitializableProcess`, and `StoppableProcess` just to make some easy interfaces for users to use for compile-time checking that they don't need to create themselves.

Typing out those names also made me think that some of these interfaces don't have consistent names, like if there's a `Stoppable` should it be `Injectable`, `Initializable` and `Processable`?

Reply:
About naming: my thoughts here were `Initializer` and `Process` are the core nouns, but these things could *also* be described by having the ability to stop running or be injected into (the core adjectives). I don't feel great about making all of them adjectives, and I'd rather have a better name for the interfaces for `Stoppable` and `Injectable` that are also nouns (if possible).

But now that I think about it, `Initializer` is also a modifier to `Process` that can stand on its own, so that's kind of odd (you can't have something that _only_ gets injected into, or _only_ stops, so those are still dependent concepts).

Reply:
I'd be in favor of changing the names of `Stoppable` and `Injectable`. I think it would be closer to the idiomatic naming like `io.Reader` or `io.Writer`. I get what you mean about `Process` being a core noun, but in general Go likes verbs for interfaces and I think that might make it easier for people to pick up. Is there a reason it couldn't just be `Processor`? I don't think that would remove the usage of the overall `Process` concept, just that it's an attribute a process can have. Maybe `Process` could be `Runner` to match the name?

Reply:
Yes, you're right. I blame my brain for being frazzled at the end of the day. :D

Maybe instead of `Inject` it becomes `Receive` and `Receiver` or something similar?

Reply:
@aphistic Sorry for dragging my feet on this.

Updated to: `Injecter`, `Initializer`, `Runner`, `Stopper`, and `Finalizer`. Let me know which combination interfaces you think would be most useful (I don't think we should necessarily support every combination for convenience).

I'm still using "process" to refer to the general idea of an "initializer or a runner (with possible injection, finalization, and graceful exit)" and the inline docs should reflect that. Please let me know if I missed a reference to the old names.